File: ga/4omini_result_2024_7/result/1968C_mini/40_1968C.c
use std::io::{self, BufRead};

const N: usize = 500;
const A: usize = 1_000_000_000;

fn proc() {
    let stdin = io::stdin();
    let input = stdin.lock().lines().next().unwrap().unwrap();
    let n: usize = input.trim().parse().unwrap();
    
    let mut x = [0; N];
    let mut a = [0; N];

    // Reading input for x
    let second_line = stdin.lock().lines().next().unwrap().unwrap();
    for (i, value) in second_line.split_whitespace().enumerate() {
        x[i + 1] = value.trim().parse().unwrap(); // Align with 1..n indexing
    }

    // Initialize a[0]
    a[0] = x[1] + 1;

    // Calculate the rest of array a
    for i in 1..n {
        for j in 0..=A {
            a[i] = x[i] + j * a[i - 1];
            if (if i + 1 < n { x[i + 1] } else { 1 }) < a[i] && a[i] != a[i - 1] {
                break;
            }
        }
    }

    // Output the results for array a
    for i in 0..n {
        print!("{} ", a[i]);
    }
    println!();
}

fn main() {
    let stdin = io::stdin();
    let handle = stdin.lock();
    let t: usize = handle.lines().next().unwrap().unwrap().trim().parse().unwrap();

    for _ in 0..t {
        proc();
    }
}

File: ga/glm4_result_2024_6/result/1968C_glm/40_1968C.rs
const N: usize = 500;
const X: usize = 500;
const A: i32 = 1000000000;

fn proc() {
    let mut n = 0;
    let mut x = [0; N];
    let mut a = [0; N];

    // Read input
    std::io::stdin().read_line(&mut String::new()).unwrap();
    std::io::stdin().read_line(&mut String::new()).unwrap();
    n = std::io::stdin().read_line(&mut String::new()).unwrap().trim().parse::<usize>().unwrap();
    for i in 1..n {
        x[i] = std::io::stdin().read_line(&mut String::new()).unwrap().trim().parse::<i32>().unwrap();
    }

    // Calculate sequence
    a[0] = x[1] + 1;
    for i in 1..n {
        let mut j = 0;
        while j <= A {
            a[i] = x[i] + j * a[i - 1];
            if (i + 1 < n && x[i + 1] < a[i]) || (i + 1 == n && 1 < a[i]) {
                break;
            }
            j += 1;
        }
    }

    // Output result
    for i in 0..n {
        print!("{} ", a[i]);
    }
    println!();
}

fn main() {
    let mut t = 0;
    std::io::stdin().read_line(&mut String::new()).unwrap();
    t = std::io::stdin().read_line(&mut String::new()).unwrap().trim().parse::<i32>().unwrap();
    for _ in 0..t {
        proc();
    }
}